test/integration/particle_emitter.cc

```diff
@@ -0,0 +1,170 @@
+                // TODO(anyone) add material check here
+                EXPECT_DOUBLE_EQ(10.0, _emitter->Data().min_velocity());
+                EXPECT_DOUBLE_EQ(20.0, _emitter->Data().max_velocity());
+                EXPECT_EQ(math::Color::Blue,
+                    msgs::Convert(_emitter->Data().color_start()));
+                EXPECT_EQ(math::Color::Green,
+                    msgs::Convert(_emitter->Data().color_end()));
+                EXPECT_DOUBLE_EQ(10.0, _emitter->Data().scale_rate());
+
+                // color range image is empty because the emitter system
+                // will not be able to find a file that does not exist
+                // TODO(anyone) this should return  "/path/to/dummy_image.png"
+                // and let rendering do the findFile instead
+                EXPECT_EQ(std::string(),
+                    _emitter->Data().color_range_image());
+              }
+              else
+              {
+                updateDefaultChecked = true;
+
+                EXPECT_TRUE(_name->Data().find(std::to_string(_entity))
+                    != std::string::npos);
+                EXPECT_EQ(_name->Data(), _emitter->Data().name());
+                EXPECT_EQ(msgs::ParticleEmitter_EmitterType_POINT,
+                    _emitter->Data().type());
+                EXPECT_EQ(math::Pose3d(0, 0, 0, 0, 0, 0), _pose->Data());
+                EXPECT_EQ(_pose->Data(),
+                    msgs::Convert(_emitter->Data().pose()));
+                EXPECT_EQ(math::Vector3d(1, 1, 1),
+                    msgs::Convert(_emitter->Data().size()));
+                EXPECT_DOUBLE_EQ(10.0, _emitter->Data().rate());
+                EXPECT_DOUBLE_EQ(0.0, _emitter->Data().duration());
+                EXPECT_FALSE(_emitter->Data().emitting());
+                EXPECT_EQ(math::Vector3d(1, 1, 1),
+                    msgs::Convert(_emitter->Data().particle_size()));
+                EXPECT_DOUBLE_EQ(5.0, _emitter->Data().lifetime());
+                // TODO(anyone) add material check here
+                EXPECT_DOUBLE_EQ(1.0, _emitter->Data().min_velocity());
+                EXPECT_DOUBLE_EQ(1.0, _emitter->Data().max_velocity());
+                EXPECT_EQ(math::Color::White,
+                    msgs::Convert(_emitter->Data().color_start()));
+                EXPECT_EQ(math::Color::White,
+                    msgs::Convert(_emitter->Data().color_end()));
+                EXPECT_DOUBLE_EQ(1.0, _emitter->Data().scale_rate());
+                EXPECT_EQ("", _emitter->Data().color_range_image());
+              }
+
+              return true;
+            });
+      });
+
+  this->server->AddSystem(testSystem.systemPtr);
+  this->server->Run(true, 1, false);
+
+  EXPECT_TRUE(updateCustomChecked);
+  EXPECT_TRUE(updateDefaultChecked);
+}
+
+/////////////////////////////////////////////////
+/// Main
+int main(int _argc, char **_argv)
+{
+  ::testing::InitGoogleTest(&_argc, _argv);
+  return RUN_ALL_TESTS();
+}
```

Comment:
Do we need this? I don't think we use this in other integration tests.

Reply:
It's not required. I removed it in 8a1db3d3012bfdcfb673c266c998952d77475716